### ISSUE TITLE
Feature: extend inputs.nginx module to provide error counter

### ISSUE THREAD

**Issue by winem** [influxdata/telegraf#2841]
### Proposal:
Extend the statistics provided by the inputs.nginx module by an error counter. 

### Current behavior:
Unfortunately, the nginx status page does not provide the number of errors. So you have to do the math manually: accepts - handled

The result is the number of errors when handling an accepted connection. 

### Desired behavior:
Calculate the number of errors in the inputs plugin itself. My fix would look like this: 

``` 
nginx.go 
...
113         // Calculating the error counter
114         errors := accepts - handled
115 
...
140                 "errors":   errors,
...
``` 

The test file needs to be extended as well, of course. 

I would love to provide a merge request but actually I'm not able to build telegraf without errors and I won't push any untested code - see #2840 

Please let me know what you think about this feature request. 

### Use case: [Why is this important (helps with prioritizing requests)]
You do not have to calculate the errors by a tick script, a CQ or on your monitoring instance itself. This saves a lot of redundant code if you use different scripts and tools evaluating those measurements.

**Comment by danielnelson** [comment]
Is there any reason not to do this calculation when performing the query?  You could do something like this:
```
select accepts-handled as errors from nginx where time > now() - 1m
```

**Comment by winem** [comment]
TL;DR: There is no compelling reason from the technical perspective. 

I just think that it would increase the usability. Without this, you would have to do the calculation in the grafana query (this would increase the complexity for already complex queries) for graphs, in kapacitor for alert handling for example, and so on. The user won't have to care about it, if we move it into the plugin itself.

**Comment by danielnelson** [comment]
I think in this case we should require the calculation to be done in the query.

**Comment by phemmer** [comment]
Yes this is closed, but I just wanted to throw in a technical argument against doing things like this: It results in storing more data in the database. Yes it's a trivial amount of data, but lots of telegraf plugins have fields which could be used to create a meaningful pre-calculated value (unfortunately there are a few plugins which actually do this, and I have my telegraf configured to drop those fields). The storage used by all these fields adds up pretty quick, especially when you have lots of systems being monitored at a high polling frequency. But calculating them costs negligible CPU time, especially when compared to disk access times.

**Comment by winem** [comment]
@phemmer Thanks for the insight and your point of view. Totally got that. 
Thanks to both of you for looking into it.
